src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -534,11 +534,12 @@ OverlayImpl::onPrepare()
             {
                 if (addr.port () == 0)
                 {
-                    Throw<std::runtime_error> ("Port not specified for "
-                        "address:" + addr.to_string ());
+                    ips.push_back (to_string (addr.at_port(51235)));
```

Comment:
Please don't make a port in the dynamic/private ports range the default.

Reply:
51235 is already the default port (see existing instances a few lines above). This patch is just to automatically set it on ips & fixed ips when not specified in the config.

Reply:
There currently is no default port in `rippled`, this is just the one these instances use.

You are suggesting to make a port from a port range that is supposed to be dynamic/private to be the default port of `rippled`.

Reply:
OK I see what you're saying, though I would argue that 51235 is currently the default port in all but name, afterall it is the default port set in the config & used by most instances (that I know of).

What do you suggest as far as default port assignment, or are you saying that #2861  is a non-issue?

Reply:
I'd recommend just grabbing an unassigned port from the proper range and maybe in parallel also asking officially for an assigned port - it is actually not that difficult or impossible to get that: https://tools.ietf.org/html/rfc6335#page-16

Reply:
I am supportive of registering a port. Give me a day, I have a plan :smiling_imp:

Reply:
Update on the plan I hinted at above:

There already is a port ([2459](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=Schwartz#David_Schwartz)) that @JoelKatz has registered; we could repurpose it for this, and make it the new default for `rippled`. We'll follow-up with IANA about this.

@movitto, it may be a good idea to hold off on this PR so that we don't add an "implicit default" that people end up relying on only to then change it. Let's wait and see whether we'll go with 2459 or register something else.

Reply:
> @movitto, it may be a good idea to hold off on this PR so that we don't add an "implicit default" that people end up relying on only to then change it. Let's wait and see whether we'll go with 2459 or register something else.

Sounds good @nbougalis, let us know how the IANA follow-up goes and I can update this PR from there.